runtime/doc/dev_tools.txt

```diff
@@ -0,0 +1,192 @@
+*dev_tools.txt*          Nvim
+
+
+                            NVIM REFERENCE MANUAL
+
+
+Nvim style guide                                        *dev-tools*
+
+The following advice is helpful when debugging Nvim itself. See also
+|debug.txt| for advice that applies to Vim.
```

Comment:
Can revisit later, but just FYI: I see this document as a place where we can document more than just debugger stuff.

Reply:
Such as? (I'd be happy to generalize this already if I have some idea.)

Reply:
Perhaps some minor hints about development setup / LSP ?, Some parts of `src/nvim/README.md`, such as ASAN, getting a stacktrace, etc.

For codebase-related info (data structures, usage of various modules, etc.) we'll probably need a separate `dev_xx.txt` file, or we could put that in here too.

Reply:
Agreed; that is the goal. I've left it at a vague "working on Nvim and debugging issues" for now (and updated the description in `:h help`).